src/shogun/multiclass/tree/RRCARTree.cpp

```diff
@@ -0,0 +1,143 @@
+ *     this list of conditions and the following disclaimer.
+ *
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *
+ *  3. Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived from
+ *     this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Written (W) 2016 Saurabh Mahindre
+ */
+
+#include <shogun/mathematics/Math.h>
+#include <shogun/mathematics/linalg/linalg.h>
+#include <shogun/mathematics/eigen3.h>
+#include <shogun/multiclass/tree/RRCARTree.h>
+
+using namespace Eigen;
+using namespace shogun;
+
+CRRCARTree::CRRCARTree(): CRandomCARTree()
+{
+	SG_ADD(&m_rotation_matrix, "m_rotation_matrix", "rotation matrix", MS_NOT_AVAILABLE);	
+}
+
+CRRCARTree::~CRRCARTree()
+{
+}
+
+bool CRRCARTree::train_machine(CFeatures* data)
+{
+	REQUIRE(data,"Data required for training\n")
+	REQUIRE(data->get_feature_class()==C_DENSE,"Dense data required for training\n")
+
+	int32_t num_features=(dynamic_cast<CDenseFeatures<float64_t>*>(data))->get_num_features();
+	int32_t num_vectors=(dynamic_cast<CDenseFeatures<float64_t>*>(data))->get_num_vectors();
+
+	SGMatrix<float64_t> feats = (dynamic_cast<CDenseFeatures<float64_t>*>(data))->get_feature_matrix();
+	m_rotation_matrix = generate_rotation_matrix(num_features);
+
+	Map<MatrixXd> map_feats (feats.matrix, feats.num_rows, feats.num_cols);
+	Map<MatrixXd> map_rotation_matrix (m_rotation_matrix.matrix, num_features, num_features);
+
+	SGMatrix<float64_t> new_mat = SGMatrix<float64_t>(num_features, num_vectors);
+	Map<MatrixXd> map_new_mat(new_mat.matrix, num_features, num_vectors);
+ 
+	MatrixXd temp = map_feats.transpose()*map_rotation_matrix;
+	map_new_mat = temp.transpose();
+	CFeatures* new_data = new CDenseFeatures<float64_t>(new_mat);
+	bool result = CRandomCARTree::train_machine(new_data);
+
+	SG_UNREF(new_data);
+	return result;
+}
+
+SGMatrix<float64_t> CRRCARTree::generate_rotation_matrix(int32_t n)
```

Comment:
Oh I was wondering why there's no matrix ;D
But yeah it can infer the backend with `SGMatrix` in param list. So it would be just like other operations. 
I didn't see matrix rotation methods in `ViennaCL` with a quick search. So we have to start from scratch and maybe use kernels to do the rotation...?